addon/decorators/key-responder.js

```diff
@@ -0,0 +1,66 @@
+import { inject as service } from '@ember/service';
+
+function populateKeyboardHandlers(responder) {
+  responder.keyboardHandlers = responder.keyboardHandlers || {};
+  if (!responder.keyboardHandlerNames) {
+    responder.keyboardHandlerNames = {};
+    for (let propertyName in responder) {
+      let propertyValue = responder[propertyName];
+      if ((typeof propertyValue === 'function') && propertyValue._emberKeyboardOnKeyDecoratorData) {
+        for (let listenerName of propertyValue._emberKeyboardOnKeyDecoratorData.listenerNames || []) {
+          responder.keyboardHandlerNames[listenerName] = propertyName;
+        }
+      }
+    }
+  }
+  for (let [listenerName, methodName] of Object.entries(responder.keyboardHandlerNames || {})) {
+    responder.keyboardHandlers[listenerName] = responder[methodName].bind(responder);
+  }
+}
+
+export default function keyResponder(DecoratedClass, opts = {}) {
+  if (opts.priority === undefined) {
+    opts.priority = 0;
+  }
+
+  if (opts.activated === undefined) {
+    opts.activated = true;
+  }
+
+  return class ClassAsKeyResponder extends DecoratedClass {
```

Comment:
this should probably have a `static name = `${DecoratedClass.name}WithKeyResponder`;

Reply:
Thanks!